instrumentation/jdbc/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jdbc/test/JdbcInstrumentationTest.java

```diff
@@ -0,0 +1,1387 @@
+            "hsqldb",
+            cpDatasources.get("tomcat").get("hsqldb").getConnection(),
+            "SA",
+            "CREATE TABLE PUBLIC.S_HSQLDB_TOMCAT (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE PUBLIC.S_HSQLDB_TOMCAT",
+            "hsqldb:mem:",
+            "PUBLIC.S_HSQLDB_TOMCAT"),
+        Arguments.of(
+            "h2",
+            cpDatasources.get("hikari").get("h2").getConnection(),
+            null,
+            "CREATE TABLE S_H2_HIKARI (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE jdbcunittest.S_H2_HIKARI",
+            "h2:mem:",
+            "S_H2_HIKARI"),
+        Arguments.of(
+            "derby",
+            cpDatasources.get("hikari").get("derby").getConnection(),
+            "APP",
+            "CREATE TABLE S_DERBY_HIKARI (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE jdbcunittest.S_DERBY_HIKARI",
+            "derby:memory:",
+            "S_DERBY_HIKARI"),
+        Arguments.of(
+            "hsqldb",
+            cpDatasources.get("hikari").get("hsqldb").getConnection(),
+            "SA",
+            "CREATE TABLE PUBLIC.S_HSQLDB_HIKARI (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE PUBLIC.S_HSQLDB_HIKARI",
+            "hsqldb:mem:",
+            "PUBLIC.S_HSQLDB_HIKARI"),
+        Arguments.of(
+            "h2",
+            cpDatasources.get("c3p0").get("h2").getConnection(),
+            null,
+            "CREATE TABLE S_H2_C3P0 (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE jdbcunittest.S_H2_C3P0",
+            "h2:mem:",
+            "S_H2_C3P0"),
+        Arguments.of(
+            "derby",
+            cpDatasources.get("c3p0").get("derby").getConnection(),
+            "APP",
+            "CREATE TABLE S_DERBY_C3P0 (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE jdbcunittest.S_DERBY_C3P0",
+            "derby:memory:",
+            "S_DERBY_C3P0"),
+        Arguments.of(
+            "hsqldb",
+            cpDatasources.get("c3p0").get("hsqldb").getConnection(),
+            "SA",
+            "CREATE TABLE PUBLIC.S_HSQLDB_C3P0 (id INTEGER not NULL, PRIMARY KEY ( id ))",
+            "CREATE TABLE PUBLIC.S_HSQLDB_C3P0",
+            "hsqldb:mem:",
+            "PUBLIC.S_HSQLDB_C3P0"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("statementUpdateStream")
+  @DisplayName(
+      "statement update on #system with #connection.getClass().getCanonicalName() generates a span")
+  void testStatementUpdate(
+      String system,
+      Connection connection,
+      String username,
+      String query,
+      String spanName,
+      String url,
+      String table)
+      throws SQLException {
+    Statement statement = connection.createStatement();
+    cleanup.deferCleanup(statement);
+    cleanup.deferCleanup(connection);
+    String sql = connection.nativeSQL(query);
+    testing.runWithSpan("parent", () -> statement.execute(sql));
```

Comment:
in the groovy test, [this is also an assertion](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/35ea26d1a3bdd8a628309a863c99c03fed4dfe7c/instrumentation/jdbc/javaagent/src/test/groovy/JdbcInstrumentationTest.groovy#L396)

```suggestion
    testing.runWithSpan("parent", () -> assertThat(statement.execute(sql)).isFalse());
```